docs/deployment/add-ons/profile-iam/README.md

```diff
@@ -0,0 +1,131 @@
+# AWS IAM for Kubeflow Profiles
+
+## Background
+
+### Kubeflow Profiles
+
+From the [official Kubeflow documentation](https://www.kubeflow.org/docs/components/multi-tenancy/overview/), a `Profile` is "a unique configuration for a user, which determines their access privileges and is defined by the Administrator". Additional details can be found [here](https://github.com/kubeflow/kubeflow/tree/d43aee19ad8b556273a627ad34367e7e56b5c97e/components/profile-controller).
+
+Kubeflow uses Profiles as a Kubernetes access control interface (e.g. role based access control [RBAC]) to "control all policies, roles, and bindings involved, and to guarantee consistency". Resources belonging to that Profile are contained within the `Profile Namespace`.
+
+To control the access by Profile users to resources outside of Kuberenetes, `Profile Plugins` have been developed to interface with the Identity and Access Management (IAM) services that manage permissions for those external resources. 
+
+The `AwsIamForServiceAccount` plugin is such a Profile plugin which allows enforcing AWS IAM access control for Profile users in order to grant/limit access to AWS resources and services.
+
+### The Profiles Controller and the `AwsIamForServiceAccount` Plugin
+
+In order for the profile controller to interface with AWS IAM access control, the profiles controller pod must be recognized as an entity that can interface with IAM and be given the necessary permissions. This is done by using IAM Role Service Accounts (IRSA).
+
+In AWS EKS, [IRSA](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) is a feature that allows enforcing AWS IAM permissions boundaries at the Kubernetes pod level. In a nutshell, a Kubernetes service account (SA) is associated with an IAM role with a role policy that scopes the IAM permissions (e.g. S3 read/write access, etc.). When a pod in the SA namespace is annotated with the SA name, EKS injects the necessary (temporary) credentials so that the pod can make requests to AWS services within the scope of role policy associated with the IRSA.
+
+In order to enforce IAM permissions boundaries at the profile level, the profiles controller IRSA itself is not given the permissions required by the profiles. Instead, admins will create an IAM role for each profile and the profile controller IRSA will have permissions to access the admin created profile role and modify it to grant the `default-editor` SA in the profile namespsace permissions to assume the role. The `default-editor` SA in the profile namespace will also be annotated with the role arn. By doing both, EKS will then inject the necessary temporary credentials into the pod in the profile namespace, granting it the permissions defined in the profile role policy.
```

Comment:
needs small reword here, EKS webhook only injects the ROLE_ARN and a token which can be used by the AWS SDK to get credentials

https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/ this will be helpful, there is a diagram at the end

same thing mentioned in above para as well

Reply:
done